proof/refine/X64/Schedule_R.thy

```diff
@@ -150,30 +150,12 @@ lemma switchLocalFpuOwner_corres[corres]:
    apply (auto simp: current_fpu_owner_Some_tcb_at fpuOwner_asrt_cross)
   done
 
-(*FIXME FPU: move tcbFlag lemmas*)
-lemma tcbFlagToWord_bit:
-  "\<exists>n<64. tcbFlagToWord flag = bit n"
-  by (auto simp: tcbFlagToWord_def ex_nat_less_eq split: tcb_flag.splits simp del: bit_def)
-
-lemma FpuDisabled_in_tcbFlagMask[simp]:
-  "FpuDisabled \<in> word_to_tcb_flags tcbFlagMask"
-  by (clarsimp simp: word_to_tcb_flags_def tcbFlagToWord_def tcbFlagMask_def Kernel_Config.config_HAVE_FPU_def)
-
-lemma tcbFlagToWord_tcbFlagMask:
-  "flag \<in> word_to_tcb_flags tcbFlagMask \<Longrightarrow> tcbFlagToWord flag && tcbFlagMask = tcbFlagToWord flag"
-  by (cut_tac flag=flag in tcbFlagToWord_bit) (fastforce simp: word_to_tcb_flags_def word_eq_iff)
-
-lemma isFlagSet_set:
-  "flag \<in> word_to_tcb_flags tcbFlagMask \<Longrightarrow> isFlagSet flag flags = (flag \<in> word_to_tcb_flags flags)"
-  apply (drule tcbFlagToWord_tcbFlagMask)
-  by (clarsimp simp: isFlagSet_def word_to_tcb_flags_def word_bw_lcs intro!: eq_eqI word_bw_comms)
-
 lemma lazyFpuRestore_corres[corres]:
   "corres dc (pspace_aligned and pspace_distinct and valid_cur_fpu and tcb_at t) \<top>
              (lazy_fpu_restore t) (lazyFpuRestore t)"
   unfolding lazy_fpu_restore_def lazyFpuRestore_def
   by (corres corres: threadGet_corres[where r="\<lambda>flags flags'. flags = word_to_tcb_flags flags'"]
-          term_simp: tcb_relation_def isFlagSet_set
+          term_simp: tcb_relation_def Kernel_Config.config_HAVE_FPU_def
```

Comment:
is this future-proof? we're unfolding a kernel config; I guess the function only exists when there is an FPU?

Reply:
Yeah, something brief to that effect would be good.